metric_learn/lmnn.py

```diff
@@ -191,10 +191,12 @@ def _loss_grad(self, X, L, dfG, impostors, it, k, reg, target_neighbors, df,
     # do the gradient update
     assert not np.isnan(df).any()
     G = dfG * reg + df * (1 - reg)
+
+    grad = 2 * L.dot(G)
```

Comment:
I put the multiplication by 2 here so that we can test the real gradient using `check_grad`